python_modules/libraries/dagster-embedded-elt/dagster_embedded_elt/sling/resources.py

```diff
@@ -355,24 +361,40 @@ def sync(
 
             yield from self._exec_sling_cmd(cmd, encoding=encoding)
 
-    @public
     def replicate(
         self,
         *,
-        replication_config: SlingReplicationParam,
-        dagster_sling_translator: DagsterSlingTranslator,
+        context: Union[OpExecutionContext, AssetExecutionContext],
+        replication_config: Optional[SlingReplicationParam] = None,
+        dagster_sling_translator: Optional[DagsterSlingTranslator] = None,
         debug: bool = False,
     ) -> Generator[MaterializeResult, None, None]:
         """Runs a Sling replication from the given replication config.
 
         Args:
             replication_config: The Sling replication config to use for the replication.
-            dagster_sling_translator: The translator to use for the replication.
+            context: Asset or Op execution context.
             debug: Whether to run the replication in debug mode.
 
         Returns:
             Optional[Generator[MaterializeResult, None, None]]: A generator of MaterializeResult
         """
+        # attempt to retrieve params from asset context if not passed as a parameter
+        if not (replication_config or dagster_sling_translator):
+            metadata_by_key = context.assets_def.metadata_by_key
+            first_asset_metadata = next(iter(metadata_by_key.values()))
+            dagster_sling_translator = first_asset_metadata.get(METADATA_KEY_TRANSLATOR)
+            replication_config = first_asset_metadata.get(METADATA_KEY_REPLICATION_CONFIG)
+
+        # if translator has not been defined on metadata _or_ through param, then use the default constructor
+        if not dagster_sling_translator:
+            dagster_sling_translator = DagsterSlingTranslator()
+
+        if replication_config is None:
+            raise Exception(
+                f"`ReplicationConfig` must be defined on metadata at {METADATA_KEY_REPLICATION_CONFIG}"
```

Comment:
This is an odd error message, since the user isn't explicitly setting the metadata -- the integration implementation is. So this message isn't actionable.

This should be handled in `validate_replication` anyways (there's a check there)